.tekton/tasks/update-repository.yaml

```diff
@@ -0,0 +1,192 @@
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "appstudio, hacbs"
+  name: update-repository
+spec:
+  description: |
+    Clones a repository, runs script in 'SCRIPT' parameter, and generates a pull-request to the repository if a change is detected.
+  params:
+    - name: COMMIT_BRANCH
+      description: Name of the branch which holds the update created by the Pipeline
+    - name: TARGET_BRANCH
+      description: Name of the branch which is modified by the Pipeline
+    - name: TARGET_GH_NAME
+      description: Name of the repository which is modified by the Pipeline
+    - name: TARGET_GH_OWNER
+      description: Owner of the repository which is modified by the Pipeline
+    - name: TARGET_GH_URL
+      description: URL of github repository which is modified by the Pipeline
+    - name: SCRIPT_IMAGE
+      description: Image reference used to execute the script
+    - name: SCRIPT_PATH
+      description: Path to the script updating the repository
+    - name: SCRIPT_ARGS
+      description: Arguments to the bash script
+      type: array
+    - name: shared-secret
+      default: infra-deployments-pr-creator
+      description: secret in the namespace which contains private key for the GitHub App
+    - name: GITHUB_APP_ID
+      description: ID of Github app used for updating PR
+      default: "305606"
+    - name: GITHUB_APP_INSTALLATION_ID
+      description: Installation ID of Github app in the organization
+      default: "35269675"
+    - name: GIT_IMAGE
+      description: Image reference containing the git command
+      default: registry.redhat.io/openshift-pipelines/pipelines-git-init-rhel8:v1.8.2-8@sha256:a538c423e7a11aae6ae582a411fdb090936458075f99af4ce5add038bb6983e8
+    - name: GIT_USER
+      description: Username to appear in the commit
+      default: "Tekton CI"
+    - name: GIT_EMAIL
+      description: Email to appear in the commit
+      default: "tekton-ci@example.com"
+  volumes:
+    - name: infra-deployments-pr-creator
+      secret:
+        # 'private-key' - private key for Github app
+        secretName: $(params.shared-secret)
+  steps:
+    - name: git-clone-repository
+      image: $(params.GIT_IMAGE)
+      workingDir: $(workspaces.workdir.path)
+      env:
+        - name: TARGET_BRANCH
+          value: $(params.TARGET_BRANCH)
+        - name: TARGET_GH_NAME
+          value: $(params.TARGET_GH_NAME)
+        - name: TARGET_GH_OWNER
+          value: $(params.TARGET_GH_OWNER)
+        - name: TARGET_GH_URL
+          value: $(params.TARGET_GH_URL)
+      script: |
+        WORK_DIR="${PWD}/${TARGET_GH_OWNER}/${TARGET_GH_NAME}"
+        if [ -e "${WORK_DIR}" ]; then
+          echo "Clean checkout of '${TARGET_GH_URL}/${TARGET_BRANCH}' in '${WORK_DIR}'"
+          cd "${WORK_DIR}"
+          git clean -d --force
+          git reset --hard
+          git checkout "${TARGET_BRANCH}"
+        else
```

Comment:
This block appears unnecessary. we can create volume by using  [emptyDir](https://github.com/redhat-appstudio/build-definitions/blob/main/task/update-infra-deployments/0.1/update-infra-deployments.yaml#L38-L44)  in task. The reason is that 1. it doesn't need to share data/files between tasks. 2. each task creates isolated volume storage if using `emptyDir`

Reply:
The task may (will) be called multiple times in the same pipeline. The shared volume will prevent having to clone the repository each time.